org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -800,6 +807,7 @@ public Preferences() {
 		autobuildEnabled = true;
 		completionEnabled = true;
 		postfixCompletionEnabled = true;
+		completionMatchCaseMode = CompletionMatchCaseMode.OFF;
```

Comment:
The default value in VS Code is auto - https://github.com/redhat-developer/vscode-java/pull/2834/commits/28360cb02bc744216d7368bea6a5b528d6da1d6b#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R588

Reply:
AKAIU, JDTLS only accept `firstLetter` or `off`, and `auto` is a "client-only" option, not visible to JDTLS. When its value is `auto`. the vscode client has [specific logic](https://github.com/redhat-developer/vscode-java/commit/28360cb02bc744216d7368bea6a5b528d6da1d6b#diff-39b2554fd18da165b59a6351b1aafff3714e2a80c1435f2de9706355b4d32351R210) to set the value before sending it to JDTLS.

Reply:
Thank you @Eskibear for the explanation.

@snjeza We want to experiment this in Insiders only at first. The server side only have on/off status. While the client can have its own settings and maps to on/off when sending to server.